### PR TITLE
Try fix snapshot export

### PR DESCRIPTION
**Pull request by elmattic**
## Summary of changes

As I can't reproduce it on a DO droplet, this PR might not fix [our issue](https://github.com/ChainSafe/forest/issues/3309). But this should not worsen the situation.

Changes introduced in this pull request:

- Use `writer` `shutdown` after flushing
- Use `file` `sync_all` before dropping

## Reference issue to close (if applicable)

Closes

## Other information and links

https://docs.rs/tokio/latest/tokio/fs/struct.File.html#method.sync_all

## Change checklist

- [ ] I have performed a self-review of my own code,
- [ ] I have made corresponding changes to the documentation,
- [ ] I have added tests that prove my fix is effective or that my feature works (if possible),
- [ ] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by lemmih — CHANGES_REQUESTED** [review]
I don't think this code should be merged without a solid reason.

File: src/chain/mod.rs [hunk 1]
```diff
@@ -54,6 +54,7 @@ pub async fn export<D: Digest>(
 
     // Flush to ensure everything has been successfully written
     writer.flush().await.context("failed to flush")?;
+    writer.shutdown().await.context("failed to shutdown")?;
```
Comment (lemmih) [inline src/chain/mod.rs:57]: This code literally does nothing. `shutdown` is only defined on sockets.

**Comment by lemmih** [comment]
I'm fairly certain the errors we've been seeing have nothing to do with the file being truncated. The magic number we add at the end of the CAR.forest.zst file is still there.

File: src/chain/mod.rs (hunk 1 above)
Comment (elmattic) [reply src/chain/mod.rs:57]: Reading the code I was thinking this will should call internally `poll_flush` method from `AsyncWrite` trait in `BufWriter`.

**Comment by elmattic** [comment]
> I don't think this code should be merged without a solid reason.

Putting the PR back to draft then.

**Comment by elmattic** [comment]
Closing it as this is probably not the right fix.

**Pull request closed without merging**